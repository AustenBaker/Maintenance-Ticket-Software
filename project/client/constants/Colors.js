const tintColor = '#2f95dc';

export default {
  tintColor,
  tabIconDefault: '#ccc',
  tabIconSelected: tintColor,
  tabBar: '#000',
  errorBackground: 'red',
  errorText: '#fff',
  warningBackground: '#EAEB5E',
  warningText: '#666804',
  noticeBackground: tintColor,
  noticeText: '#fff',
};
