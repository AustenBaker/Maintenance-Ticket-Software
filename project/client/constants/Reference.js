// REGEX constant 
export const REGEX = {
    // name pattern: up to 32 ASCII alphabetical characters
    NAME: /^[a-zA-Z]{1,32}$/,
    // email pattern:  some(.name)*@site.com
    // where some name & site are any combination of letters
    // and numbers, and the site ends in a 2-3 letter/digit extension
    EMAIL: /^[[\w]+[\.\w]*@[\w]+\.[\w]{2,3}]{6,32}$/,
    // phone pattern:  ###-###-#### where # is a digit 0-9
    PHONE: /^[\d]{3}-[\d]{3}-[\d]{4}$/,
    // password pattern: a-zA-Z0-9_-!@#$%^&*+=
    // requires at least one each: uppercase letter, number, punctuation
    // length of 8-32 characters
    PASSWORD: /^[[\w]*[A-Z]+[\d]+[\-!@#$%\^\*\+=]+]{8,32}$/
}


export const USER_TYPE = ['resident','maintenance','management'];
export const ENTRY_PERMISSION = ['accompanied','notify','any'];
export const PREFERRED_CONTACT = ['email','text'];

