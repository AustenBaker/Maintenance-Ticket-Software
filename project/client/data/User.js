import * as React from 'react';
import { Input, Text, Picker } from 'react-native';
import * as CONSTANTS from '../constants/Reference';

export default class User extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            first_name: props.first_name,
            last_name: props.last_name,
            units: [...props.units],
            email: props.email,
            phone: props.phone,
            contact: props.contact,
            entry_permission: props.entry_permission,
            user_type: props.user_type,
            note: props.note,
            edit_mode: props.edit_mode,
            tickets: []
        }
    }

    /**
     * Declare default values for User props in the event they
     * are not all passed into the constructor.
     */
    static defaultProps = {
        first_name: "John",
        last_name: "Doe",
        units: ["1703"],
        email: "default@CastlebergCommunities.com",
        phone: "000-000-0000",
        contact: "email",
        entry_permission: "accompanied",
        user_type: "resident",
        note: "",
        edit_mode: "false",
        tickets: []
    }

    /**
     * This method generates the <View> to display the
     * user information.
     * 
     * @returns React Native encoding for user display
     */
    displayUser = () => {
        var content;
        
        // if phone number exists, create text output
        // and add star to user's preferred contact type
        var contact = this.state.email;
        if (this.state.phone !== this.defaultProps.phone) {
            if (this.state.contact === 'email') {
                contact += "* Phone: ";
                contact += this.state.phone;
            } else {
                contact += " Phone: ";
                contact += this.state.phone;
                contact += "*";
            }
            contact += "\n* = Preferred contact method."
        }

        // create <Text> container for entry permission data
        var entry = "";
        if (this.state.entry_permission === "any") {
            entry = (
                <Text>
                    Entry: Allowed
                </Text>
            );
        } else if (this.state.entry_permission === "notify") {
            entry = (
                <Text>
                    Entry: Notify before entry.
                </Text>
            );
        } else entry = (
            <Text>
                Entry: Accompanied entry only.
            </Text>
        );

        // if note exists, create a <Text> container for it
        var note = "";
        if (this.state.note !== "") {
            note = (
                <Text>
                  {"Note: \n"}
                  {this.state.note}
                </Text>
            );
        }

        // label & put user info into a <View><Text> wrapper
        // for display
        content = (
            <View>
              <Text>
                Name:
                {this.state.first_name}
                {this.state.last_name}
              </Text>
              <Text>
                Email:
                {contact}
              </Text>
              {entry}
              {note}
            </View>
          );
        return content;
    }

    /**
     * This method is used to assign a unit to a resident user.
     * No duplicate values will be added to assigned units list.
     * 
     * @param units array of units to be assigned to user
     */
    assignUnit = (props) => {
        for (let unit of props.units) {
            // check to see if the unit is already assigned to user
            if (unit in this.state.units === 'false') {
                this.setState(units, [...this.state.units, unit]);
            }
        }
    }

    /**
     * This method updates user information.
     * Allows update for all user data except password.
     * 
     * @param first_name User first name
     * @param last_name User last name
     * @param email User email address
     * @param phone User phone number (optional)
     * @param contact User preferred contact method (email/text)
     * @param entry_permission User entry preference
     * @param note Note from user regarding special circumstances
     * 
     * @returns true if updated successfully
     */
    update = (props) => {
        var updated = false;
        var checked = [];
        // check and store valid values in an array
        if ('first_name' in props && CONSTANTS.REGEX.NAME.exec(props.first_name)) {
            checked.push({['first_name']: props.first_name});
        }
        if ('last_name' in props && CONSTANTS.REGEX.NAME.exec(props.last_name)) {
            checked.push({['last_name']: props.last_name});
        }
        if ('email' in props && CONSTANTS.REGEX.EMAIL.exec(props.email)) {
            checked.push({['email']: props.email});
        }
        if ('phone' in props && CONSTANTS.REGEX.PHONE.exec(props.phone)) {
            checked.push({['phone']: props.phone});
        }
        if ('contact' in props && props.contact in CONSTANTS.PREFERRED_CONTACT) {
            checked.push({['contact']: props.contact});
        }
        if ('entry_permission' in props &&
          props.entry_permission in  CONSTANTS.ENTRY_PERMISSION){
            checked.push({['entry']: props.entry});
        }
        if ('note' in props) { // add REGEX check here for note
            checked.push({['note']: props.note});
        }
        
        // update server state and this.setState
        return updated;
    }

    /**
     * This method allows a management user to change
     * another user's user type.
     * 
     * @requires User to be management user type
     * 
     * @param userName Name of user to be upgraded
     * @param userType Desired new user type
     * 
     * @returns True if user type successfully changed.
     */
    setUserType = (props) => {
        let success = false;
        if (this.state.user_type === CONSTANTS.USER_TYPE[2]) {
            // need to implement this!
            // access server in order to
            // verify userName is a valid user and then update
            // their user type to userType
            // once a response is received from server,
            // update success & return
            success = true;
        }
        return success;
    }

    /**
     * This method checks to see if the user is authorized to
     * perform the requested activity.
     * 
     * @param activity The activity to be performed
     * 
     * @returns True if the user is authorized to perform the
     * specified activity.
     */
    isAuthorized(props) {
        let valid = false;
        return valid;
    }

    /**
     * This method generates the form to display when editing
     * User profile.
     * 
     * @returns React Native encoding to edit user profile
     */
    editUser = () => {
        // store React Native element encoding being generated for return
        let content;

        // store user profile modifications in progress before submission
        // set initial value to pre-existing user profile
        let profile = [];
        // read out current state values into profile until
        // users submit form.
        const [contact, setContact] = useState(CONSTANTS.PREFERRED_CONTACT[0]);
        const [entry, setEntry] = useState(CONSTANTS.ENTRY_PERMISSION[0]);
        content = (
            <form name="updateProfile">
              <Input
                label="First Name"
                defaultValue={this.state.first_name}
                keyboardType="default"
                maxLength="32"
                selectTextOnFocus="true"
                textContentType="name"
                autoCompleteType="name"
                errorMessage="This field is required."
                onSubmitEditing={fname => this.setState(first_name, fname)}
              />
              <Input
                label="Last Name"
                defaultValue={this.state.last_name}
                keyboardType="default"
                maxLength="32"
                selectTextOnFocus="true"
                textContentType="familyName"
                autoCompleteType="name"
                errorMessage="This field is required."
                onSubmitEditing={lname => this.setState(last_name, lname)}
              />
              <Input
                label="Email"
                defaultValue={this.state.email}
                keyboardType="email-address"
                maxLength="32"
                selectTextOnFocus="true"
                textContentType="emailAddress"
                autoCompleteType="email"
                errorMessage="This field is required.  Please enter valid email address."
                onSubmitEditing={emailAddr => this.setState(email, emailAddr)}
              />
              <Input
                label="Phone Number"
                defaultValue={this.state.phone}
                keyboardType="phone-pad"
                maxLength="12"
                selectTextOnFocus="true"
                textContentType="telephoneNumber"
                autoCompleteType="tel"
                errorMessage="Please enter valid phone number: ###-###-####"
                onSubmitEditing={phoneNum => this.setState(phone, phoneNum)}
              />
              <Picker
                label="Preferred Contact Method"
                selectedValue={contact}
                onValueChange={(itemValue, itemIndex) => setContact(itemValue)}
                errorMessage='This field is required.'
                >
                <Picker.Item
                    label='Email'
                    value={CONSTANTS.PREFERRED_CONTACT[0]}
                />
                <Picker.Item
                    label='Text'
                    value={CONSTANTS.PREFERRED_CONTACT[1]}
                />
              </Picker>
              <Picker
                label="Entry Permission"
                selectedValue={entry}
                onValueChange={(itemValue, itemIndex) => setEntry(itemValue)}
                errorMessage='This field is required.'
                >
                <Picker.Item
                    label='Allow accompanied entry.'
                    value={CONSTANTS.ENTRY_PERMISSION[0]}
                />
                <Picker.Item
                    title='Notify before entry.'
                    value={CONSTANTS.ENTRY_PERMISSION[1]}
                />
                <Picker.Item
                    title='Allow entry.'
                    value={CONSTANTS.ENTRY_PERMISSION[2]}
                />
              </Picker>
              <Input
                  label="Note"
                  onSubmitEditing={someNote => this.setState(note, someNote)}
                  defaultValue={this.state.note}
                  keyboardType="default"
                  maxLength="256"
                  selectTextOnFocus="true"
              />
            </form>
        );
        return content;
    }

    /**
     * This method generates a list of tickets created by this
     * user.
     * 
     * @param filter Filter ticket list by none, open, closed.
     * 
     * @returns List of user's tickets.
     */
    listTickets = (props) => {
        var content = [];
        if (props.filter === undefined || props.filter === 'none') {
            content = this.state.tickets;
        } else if (props === 'open') {
            // generates list of user's tickets that are open
            for (let ticket of this.state.tickets) {
                if (ticket.isOpen()) {
                    content.push(ticket);
                }
            }
        } else if (props === 'closed') {
            // generates list of user's tickets that are closed
            for (let ticket of this.state.tickets) {
                if (!ticket.isOpen()) {
                    content.push(ticket);
                }
            }
        }
        return content;
    }

    /**
     * This method allows user to update their password.
     * 
     * @param pwd New password
     * 
     * @returns True if password change completed successfully.
     */
    changePassword = (props) => {
        var updated = false;
        if (props.pwd !== undefined && CONSTANTS.REGEX.PASSWORD.exec(props.pwd)) {
            // process password update
        }
        return updated;
    }

    /**
     * This method renders the User element on the client device.
     * 
     * @returns React Native encoding for User element display.
     */
    render = () => {
        var content;
        if (this.state.editMode===false){
            // if not in edit mode, simply display user info
            content=this.displayUser();
        } else {
            // if in edit mode, display form for user info update
            content=this.editUser();
        }
        return (content);
    }
}