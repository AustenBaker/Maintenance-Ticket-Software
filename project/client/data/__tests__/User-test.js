import * as React from 'react';
import { FormInput, Text } from 'react-native';
import renderer from 'react-test-renderer';
import { fireEvent, render, wait, getByPlaceholderText, getByDisplayValue, getByRole, getByLabelText, getByTestId, getByTitle, getByText } from '@testing-library/react-native';
import * as JestNative from '@testing-library/jest-native';
import * as CONSTANTS from '../../constants/Reference';

import User from '../UserTwo';
import { TextInput } from 'react-native-gesture-handler';

expect.extend({...JestNative});

jest.disableAutomock();

jest.mock('expo', () => ({
    AppLoading: 'AppLoading',
}));

const SAMPLE_USER = [{
    username: "frontend",
    first: "Peter",
    last: "Piper",
    units: [{
      property: CONSTANTS.PROPERTY.WSP,
      number: '1703'
    }],
    email: 'pick@peck.com',
    phone: '123-456-7890',
    contactPreference: CONSTANTS.PREFERRED_CONTACT.EMAIL,
    entryPermission: CONSTANTS.ENTRY_PERMISSION.ACC,
    type: CONSTANTS.USER_TYPE.RES,
    note: "",
    edit_mode: false,
    tickets: [],
    activate: true
  },
  {
    username: "dev",
    first: "Red",
    last: "Rover",
    units: [{
      property: CONSTANTS.PROPERTY.WSP,
      number: '1703'
    }],
    email: 'ima@yahoo.com',
    phone: '123-456-7890',
    contactPreference: CONSTANTS.PREFERRED_CONTACT.TXT,
    entryPermission: CONSTANTS.ENTRY_PERMISSION.NOT,
    type: CONSTANTS.USER_TYPE.MNT,
    note: "",
    edit_mode: false,
    tickets: [],
    activate: true
  },
  {
    username: "mgt",
    first: "Harry",
    last: "Chin",
    units: [],
    email: "ineed@shave.com",
    phone: "0987-654-321",
    contactPreference: CONSTANTS.PREFERRED_CONTACT.TXT,
    entryPermission: CONSTANTS.ENTRY_PERMISSION.ANY,
    type: CONSTANTS.USER_TYPE.MGMT,
    note: "",
    edit_mode: false,
    tickets: [],
    activate: true
  }
];

const SAMPLE_USER_PWD = [
  "12345678",
  "S0m3Secur3P@$$w0rd",
  "management"
];


describe(`User`, () => {
    jest.useFakeTimers();

    beforeEach(() => {
      // TODO: test setup
    });

    afterEach(() => {
      // TODO: test cleanup
    });


    it(`renders the user screen`, () => {
      const tree = renderer.create(<User />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it(`renders the user profile editing screen`, () => {
      const tree = renderer.create(<User edit_mode />).toJSON();
      expect(tree).toMatchSnapshot();
    });


    describe(`created with default values`, () => {
      const { getByTestId, getByTitle, getByText, baseElement } = render(<User />);

      let testElement = getByTestId(`user-name`);
      test(`renders User name`, () => {
        expect(testElement).toBeEnabled();
      });

      testElement = getByTestId(`user-first`);
      test(`renders User first name`, () => {
        expect(testElement).toBeEnabled();
      });

      testElement = getByTestId(`user-last`);
      test(`renders User last name`, () => {
        expect(testElement).toBeEnabled();
      });

      testElement = getByTestId(`user-email`);
      test(`renders User email`, () => {
        expect(testElement).toBeEnabled();
      });

      // TODO: Figure out how to test to verify
      // this item is _not_ displayed without
      // crashing the test -.-
      // testElement = getByTestId(`user-phone`);
      test.skip(`does not render User phone number`, () => {
        expect(testElement).toBeUndefined();
      });

      // TODO: Figure out how to test this in a
      // way that doesn't break the test
      // testElement = getByText(/\*/);
      test.skip(`does not render User contact preferences`, () => {
        expect(testElement).toBeUndefined();
      });

      testElement = getByTestId(`user-entry`);
      test(`renders User entry permissions`, () => {
        expect(testElement).toBeEnabled();
      });

      // TODO: figure out a method to test this
      // that doesn't break the test
      // testElement = getByTestId(`user-note`);
      test.skip(`does not render User note`, () => {
        expect(testElement).toBeUndefined();
      });

      testElement = getByTestId(`edit-button`);
      test(`renders Edit Button`, () => {
        expect(testElement).toBeEnabled();
      });

      /**
      test(``, () => {
        expect(testElement).toBeEnabled();
      });
       */
    });

    // TODO: figure out how to implement query selectors to find
    // relevant properties & values for validation tests
    describe(`created with default values in edit_mode`, () => {
      const { getByPlaceholderText, getByTestId, getByTitle, getByText, baseElement } = render(<User edit_mode/>);
      let testElement = getByPlaceholderText(`First Name`);
      test(`has the placeholder first name`, () => {
        expect(testElement).toBeEnabled();
      });

      testElement = getByPlaceholderText(`Last Name`);
      test(`has the placeholder last name`, () => {
        expect(testElement).toBeEnabled();
      });

      // TODO: figure out how to test units assigned
      // this way won't work
      test.skip(`has the placeholder unit assigned`, () => {
        expect(testElement).toHaveProperty(units);
        expect(testElement.props.units).arrayContaining([`1703`]);
        expect(testElement.props.units).toHaveLength(1);
      });

      testElement = getByPlaceholderText(`default@CastlebergCommunities.com`);
      test(`has the placeholder email address`, () => {
        expect(testElement).toBeEnabled();
      });

      testElement = getByPlaceholderText(`000-000-0000`);
      test(`has the default phone number`, () => {
        expect(testElement).toBeEnabled();
      });

      // TODO: figure out why this test is not working
      // the RCTPicker is not rendering the testId in
      // the render()???
      // testElement = getByTestId(`user-contact`);
      test.skip(`has the default contact preference`, () => {
        expect(testElement).toBeEnabled();
      });

      // testElement = getByTestId(`user-entry-w`);
      test.skip(`has the default entry permission`, () => {
        expect(testElement).toBeEnabled();
      });

      // TODO: figure out how to test this value
      // this isn't rendered and can't be tested this way
      // need an accessor somehow....
      // testElement = getByPlaceholderText(CONSTANTS.USER_TYPE.RES);
      test.skip(`has the default user type`, () => {
        expect(testElement).toBeEnabled();
      });

      // TODO: why is this also having problems?
      // it shows up in the snapshot, and the error
      // message, so what is going on?
      // testElement = getByTestId(`edit-note`);
      test.skip(`has the default user note`, () => {
        expect(testElement).toBeEnabled();
      });

      // TODO: figure out how to test this
      // this isn't rendered and can't be tested this way
      // need an accessor somehow....
      test.skip(`has no tickets`, () => {
        // expect(tree.props).toHaveProperty(tickets);
        expect(tree.props.tickets).toBeUndefined();
      });

      testElement = getByTestId(`update-button`);
      test(`has an Update Button`, () => {
        expect(testElement).toBeEnabled();
      });

      // TODO: Figure out why getBy are throwing declaration exceptions instead of
      // returning 'undefined' when target is not found
      // testElement = getByTestId(`create-button`);
      // test(`does not have a Create Account Button`, () => {
      //   expect(testElement).toBeUndefined();
      // });

      testElement = getByTestId(`reset-button`);
      test(`has a Reset Button`, () => {
        expect(testElement).toBeEnabled();
      });

      testElement = getByTestId(`cancel-button`);
      test(`has a Cancel Button`, () => {
        expect(testElement).toBeEnabled();
      });

    });

    /**
    it(``, () => {
      // TODO: Add more tests
    });

    it(``, () => {
      // TODO: Add more tests
    });
     */
  });
