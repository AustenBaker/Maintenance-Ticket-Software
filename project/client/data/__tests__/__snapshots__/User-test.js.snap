// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User renders the user profile editing screen 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    autoCompleteType="name"
    errorMessage="This field is required."
    keyboardType="default"
    label="First Name"
    maxLength={32}
    onChangeText={[Function]}
    placeholder="First Name"
    rejectResponderTermination={true}
    selectTextOnFocus={true}
    textContentType="name"
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    autoCompleteType="name"
    errorMessage="This field is required."
    keyboardType="default"
    label="Last Name"
    maxLength={32}
    onSubmitEditing={[Function]}
    placeholder="Last Name"
    rejectResponderTermination={true}
    selectTextOnFocus={true}
    textContentType="familyName"
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    autoCompleteType="email"
    errorMessage="This field is required.  Please enter valid email address."
    keyboardType="email-address"
    label="Email"
    maxLength={32}
    onSubmitEditing={[Function]}
    placeholder="default@CastlebergCommunities.com"
    rejectResponderTermination={true}
    selectTextOnFocus={true}
    textContentType="emailAddress"
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    autoCompleteType="tel"
    errorMessage="Please enter valid phone number: ###-###-####"
    keyboardType="phone-pad"
    label="Phone Number"
    maxLength={12}
    onSubmitEditing={[Function]}
    placeholder="000-000-0000"
    rejectResponderTermination={true}
    selectTextOnFocus={true}
    textContentType="telephoneNumber"
    underlineColorAndroid="transparent"
  />
  <View>
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Email",
            "textColor": undefined,
            "value": "email",
          },
          Object {
            "label": "Text",
            "textColor": undefined,
            "value": "text",
          },
        ]
      }
      onChange={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View>
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Allow accompanied entry",
            "textColor": undefined,
            "value": "accompanied",
          },
          Object {
            "label": "Notify before entry",
            "textColor": undefined,
            "value": "notify",
          },
          Object {
            "label": "Allow entry",
            "textColor": undefined,
            "value": "any",
          },
        ]
      }
      onChange={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <TextInput
    allowFontScaling={true}
    defaultValue=""
    keyboardType="default"
    label="Note"
    maxLength={255}
    onSubmitEditing={[Function]}
    rejectResponderTermination={true}
    selectTextOnFocus={true}
    testID="note-edit"
    underlineColorAndroid="transparent"
  />
  <View
    accessibilityLabel="Update Profile Button"
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="update-button"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Update
      </Text>
    </View>
  </View>
  <View
    accessibilityLabel="Reset Profile Button"
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="reset-button"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Reset
      </Text>
    </View>
  </View>
  <View
    accessibilityLabel="Cancel Button"
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="cancel-button"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Cancel
      </Text>
    </View>
  </View>
</View>
`;

exports[`User renders the user screen 1`] = `
<View>
  <Text
    testID="user-name"
  >
    Name: 
    <Text
      testID="user-first"
    >
      First Name
    </Text>
    <Text
      testID="user-last"
    >
      Last Name
    </Text>
  </Text>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 17,
      }
    }
    testID="user-email"
  >
    Email: 
    default@CastlebergCommunities.com
  </Text>
  
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 17,
      }
    }
    testID="user-entry"
  >
    Entry: Accompanied entry only.
  </Text>
  
  <View
    accessibilityLabel="Edit Profile Button"
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="edit-button"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Edit Profile
      </Text>
    </View>
  </View>
</View>
`;
